### PR TITLE
pd applications are deleted when a user is deleted

### PR DESCRIPTION
**Pull request by clareconstantine**
When a user account is deleted, we now delete their pd applications. This will prevent these [500 errors]( https://app.honeybadger.io/projects/3240/faults/36854549) which were caused by trying to load list view with applications whose user had been deleted.

### CODE REVIEW

File: dashboard/test/models/user_test.rb [hunk 1]
```diff
@@ -2005,6 +2005,20 @@ def mock_geocoder_result(result)
     assert_equal "'invalid' does not appear to be a valid e-mail address", e.message
   end
 
+  test 'deleting user deletes dependent pd applications' do
+    teacher = create :teacher
+    application = create :pd_teacher1819_application, user: teacher
+    assert_equal teacher.id, application.user_id
```
Comment (aoby) [inline dashboard/test/models/user_test.rb]: we don't need this assertion

File: dashboard/test/models/user_test.rb [hunk 2]
```diff
@@ -2005,6 +2005,20 @@ def mock_geocoder_result(result)
...
+    assert_equal application.id, teacher.pd_applications.first.id
+
+    teacher.destroy
+
+    assert teacher.reload.deleted?
+    assert_raise ActiveRecord::RecordNotFound do
+      Pd::Application::Teacher1819Application.find(application.id)
+    end
```
Comment (aoby) [inline dashboard/test/models/user_test.rb]: How about ?
```ruby
refute Pd::Application::Teacher1819Application.exists?(application.id)
```

Comment (clareconstantine) [reply dashboard/test/models/user_test.rb]: ooh I forgot about `exists?`. much nicer.

**Comment by clareconstantine** [comment]
[Applications whose user has been deleted cause 500 errors](https://trello.com/c/ZbvDrtI1/482-applications-whose-user-has-been-deleted-cause-500-errors)

**Pull request merged**